java/client/src/main/java/glide/api/RedisClusterClient.java

```diff
@@ -1072,6 +1080,14 @@ public CompletableFuture<Object[]> scan(ClusterScanCursor cursor, ScanOptions op
                         result -> new Object[] {new NativeClusterScanCursor(result[0].toString()), result[1]});
     }
 
+    @Override
+    public CompletableFuture<Object[]> scanBinary(ClusterScanCursor cursor, ScanOptions options) {
+        return commandManager
+                .submitClusterScan(cursor, options, this::handleArrayResponseBinary)
+                .thenApply(
+                        result -> new Object[] {new NativeClusterScanCursor(result[0].toString()), result[1]});
```

Comment:
I think this is fine, but I'd be curious if the Rust layer can ever return non-UTF-8 binary data.

Reply:
for the cursor...?

Reply:
yeah, I was thinking that too